workers/blescanmulti.py

```diff
@@ -0,0 +1,51 @@
+import time
+from interruptingcow import timeout
+from bluepy.btle import Scanner, DefaultDelegate
+from mqtt import MqttMessage
+import mqtt
+from workers.base import BaseWorker
+from logger import _LOGGER
+
+REQUIREMENTS = ['bluepy']
+
+class ScanDelegate(DefaultDelegate):
+  def __init__(self):
+    DefaultDelegate.__init__(self)
+
+  def handleDiscovery(self, dev, isNewDev, isNewData):
+    if isNewDev:
+      _LOGGER.debug("Discovered new device: %s" % dev.addr)
+
+class BlescanmultiWorker(BaseWorker):
+  def searchmac(self, devices, mac):
+    present = False
+    for dev in devices:
+      if dev.addr == mac.lower():
+         present = True
+
+    return present
+
+  def getrssi(self, devices, mac):
+    rssivalue = -999
+    for dev in devices:
+      if dev.addr == mac.lower():
+         rssivalue = dev.rssi
+
+    return rssivalue
+
+  def status_update(self):
+    scanner = Scanner().withDelegate(ScanDelegate())
+    devices = scanner.scan(10.0)
+    ret = []
+
+    for name, mac in self.devices.items():
+      try:
+        if self.searchmac(devices, mac):
```

Comment:
Maybe this method should return dev object, then you can get rssi from it ? And no need to duplicate logic for finding device by addr again.

Reply:
```
def searchmac(self, devices, mac):
    # present = False
    for dev in devices:
      if dev.addr == mac.lower():
         return dev

    return ??
```
For example like this?
What to return, in your opinion, if it does not find the mac address?

Reply:
```return None```

then in upper method something like this
```python
device = self.searchmac(devices, mac)
if device ...
```